okhttp/src/main/java/com/squareup/okhttp/Headers.java

```diff
@@ -172,7 +172,7 @@ public static Headers of(String... namesAndValues) {
     private final List<String> namesAndValues = new ArrayList<>(20);
 
     /** Add an header line containing a field name, a literal colon, and a value. */
-    Builder addLine(String line) {
+    public Builder add(String line) {
```

Comment:
I think we want two APIs here.

addLine() is the awkward one that handles all the edge cases that deal with broken data, like lines `:foo: bar`. We support that to deal with old bugs in the response cache.

And the new one you're adding, `add` should be more strict. Maybe enforce a non-empty key?